### PR TITLE
Integrate user authentication via Apollo Server

### PR DESCRIPTION
**Pull request by gurbirkalsi**
Apollo Server is an open-source GraphQL server that will serve as middleware for handling user authentication hooks such as login, register, etc. Apollo takes the Prisma schema context and resolves mutations to data managed by Prisma. 

User authentication involves generating tokens via `jsonwebtoken` and handling encyrption via `bcryptjs` for ensuring integrity of generated tokens.

### CODE REVIEW

File: prisma/README.md [hunk 1]
```diff
@@ -5,60 +5,30 @@ Pbench Dashboard Backend is a server side platform for storing and retrieving in
...
-├── datamodel.prisma          # schema definitions for GraphQL data model
-├── prisma.yml                # configuration for Prisma client
-├── package.json              # project dependencies
+├── src
+│   ├── generated
+│   │   ├── prisma-client
+│   │   │   ├── index.d.ts        # typescript schema definition
+│   │   │   ├── index.js          # Prisma client entrypoint
+│   │   │   └── prisma-schema.js  # generated Prisma client schema
+│   │   ├── index.js              # Apollo server entrypoint
```
Comment (portante) [inline prisma/README.md]: One level too deep?

**Review by portante — CHANGES_REQUESTED** [review]
Just a simple comment cleanup, but looks good!

File: prisma/README.md (hunk 1 above)
Comment (gurbirkalsi) [reply prisma/README.md]: Fixed!

**Review by anishaswain — APPROVED** [review]
lgtm

**Review by portante — APPROVED** [review]

**Pull request merged**